packages/core/src/core.ts

```diff
@@ -815,6 +815,9 @@ class ThresholdKey implements ITKey {
     const latestShareRes = await this.catchupToLatestShare({ shareStore: ss, includeLocalMetadataTransitions: true });
     // if not in poly id list, metadata is probably outdated
     // is !this.metadata.polyIDList.includes(latestShareRes.latestShare.polynomialID)
+    if (!latestShareRes.shareMetadata.polyIDList.find((tuple) => tuple[0] === this.metadata.getLatestPublicPolynomial().getPolynomialID())) {
```

Comment:
move this before the comment as the comment are meant for line 821

can add comment to check if the latest share is part of the current tkey instances for this new code